### PR TITLE
#940: [typescript .d.ts] Toggle (closes #940)

### LINKED ISSUE #940: [typescript .d.ts] Toggle
← #939

## requirements
{describe the new feature}

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by ascariandrea**
Closes #940

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.

- [ ] Google Chrome
- [ ] Internet Explorer

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

File: src/toggle/Toggle.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { CSSProperties, PureComponent } from 'react'
+
+export type ToggleProps = {
+  value?: boolean,
+  onChange: () => any, // TODO: not used
```
Comment (FrancescoCioria) [inline src/toggle/Toggle.d.ts]: this should be optional and with same signature as `requestChange`:
```
onChange?: (value: boolean) => void`
```

File: src/toggle/Toggle.d.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  valueLink?: {
+    value?: boolean,
+    requestChange: (value: boolean) => any
```
Comment (FrancescoCioria) [inline src/toggle/Toggle.d.ts]: it should return `void`

**Comment by ascariandrea** [comment]
Done!

File: src/toggle/Toggle.d.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { CSSProperties, PureComponent } from 'react'
+
+export type ToggleProps = {
+  value?: boolean,
+  onChange?: () => void, // TODO: not used
```
Comment (FrancescoCioria) [inline src/toggle/Toggle.d.ts]: actually it's used, `getValueLink` returns and object with a `requestChange` that can be the original `props.onChange` or `props.valueLink.requestChange`

Comment (ascariandrea) [inline src/toggle/Toggle.d.ts]: 👍

**Pull request merged**